testing/go/domain_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package _go
+
+import (
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func TestDomain(t *testing.T) {
+	RunScripts(t, []ScriptTest{
+		{
+			Name:        "create domain",
+			SetUpScript: []string{},
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:       `CREATE DOMAIN year AS integer CONSTRAINT not_null_c NOT NULL CONSTRAINT null_c  NULL;`,
+					ExpectedErr: `conflicting NULL/NOT NULL constraints`,
+				},
+				{
+					Query:       `CREATE DOMAIN year AS integer NULL NOT NULL;`,
+					ExpectedErr: `conflicting NULL/NOT NULL constraints`,
+				},
+				{
+					Query:    `CREATE DOMAIN year AS integer DEFAULT 1999 NOT NULL CONSTRAINT year_check CHECK (((VALUE >= 1901) AND (VALUE <= 2155)));`,
+					Expected: []sql.Row{},
+				},
+				{
+					Query:       `CREATE DOMAIN year AS integer CONSTRAINT year_check CHECK (((VALUE >= 1901) AND (VALUE <= 2155)));`,
+					ExpectedErr: `type "year" already exists`,
+				},
+				{
+					Query:    `CREATE DOMAIN year_with_check AS integer CONSTRAINT year_check CHECK (((VALUE >= 1901) AND (VALUE <= 2155)));`,
+					Expected: []sql.Row{},
+				},
+				{
+					Query:    `CREATE DOMAIN year_with_two_checks AS integer CONSTRAINT year_check_min CHECK (VALUE >= 1901) CONSTRAINT year_check_max CHECK (VALUE <= 2155);`,
+					Expected: []sql.Row{},
+				},
+				{
+					Query:       `CREATE TABLE test_table (id int primary key, v non_existing_domain);`,
+					ExpectedErr: `type "non_existing_domain" does not exist`,
+				},
+			},
+		},
+		{
+			Name: "create table with domain type",
+			SetUpScript: []string{
+				`CREATE DOMAIN year AS integer CONSTRAINT year_check CHECK (((VALUE >= 1901) AND (VALUE <= 2155)));`,
+			},
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:    `CREATE TABLE table_with_domain (pk int primary key, y year);`,
+					Expected: []sql.Row{},
+				},
+				{
+					Query:    `INSERT INTO table_with_domain VALUES (1, 1999)`,
+					Expected: []sql.Row{},
+				},
+				{
+					// TODO: the correct error msg: `value for domain year violates check constraint "year_check"`
+					Query:       `INSERT INTO table_with_domain VALUES (2, 1899)`,
+					ExpectedErr: `Check constraint "year_check" violated`,
```

Comment:
Both error messages have `constraint "year_check"`, so we could shorten our error check to just that. We wouldn't have to change it later that way